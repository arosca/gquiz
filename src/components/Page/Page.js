import React from 'react'
import styled from 'styled-components/native'

import { View } from 'react-native'

const PageEl = styled(View)`
  flex: 1;
  padding: 60px 30px;
`

const BodyEl = styled(View)`
  flex: 1;
  align-items: center;
  justify-content: center;
`

/**
 * Generic wrapper component used for every screen
 * for standardizing layout: alignment, margins and paddings
 */
 type Props = {
   children: mixed
 }
class Page extends React.Component<Props> {
  render() {
    return (
      <PageEl>{ this.props.children }</PageEl>
    )
  }
}

const HeadEl = styled(View)`
  height: 100px;
`


/**
 * Page header
 */
Page.head = (props: Props) => <HeadEl>{ props.children }</HeadEl>

/**
 * Body of page
 */
Page.body = (props: Props) => <BodyEl>{ props.children }</BodyEl>

export default Page
