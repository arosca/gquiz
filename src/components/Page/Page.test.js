import renderer from 'react-test-renderer'
import React from 'react'

import { Text } from 'react-native'

import Page from './Page'

it('renders without crashing', () => {
  const rendered = renderer.create(<Page />).toJSON()
  expect(rendered).toBeTruthy()
})

it('renders the children/content', () => {
  const component = renderer.create(<Page><Text>test</Text></Page>).toJSON()
  expect(component).toMatchSnapshot()
})
