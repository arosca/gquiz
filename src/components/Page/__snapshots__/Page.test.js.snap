// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the children/content 1`] = `
<View
  style={
    Array [
      Object {
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
        "paddingBottom": 60,
        "paddingLeft": 30,
        "paddingRight": 30,
        "paddingTop": 60,
      },
      undefined,
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
    test
  </Text>
</View>
`;
